src/mono/wasm/runtime/scheduling.ts

```diff
@@ -19,23 +18,26 @@ if (globalThis.navigator) {
 }
 
 function pump_message() {
-    while (timeout_queue.length > 0) {
-        --pump_count;
-        const cb: Function = timeout_queue.shift()!;
-        cb();
-    }
-    while (pump_count > 0) {
-        --pump_count;
+    if (pump_count > 0) {
+        pump_count = 0;
         cwraps.mono_background_exec();
     }
 }
 
-function mono_wasm_set_timeout_exec(id: number) {
-    cwraps.mono_set_timeout_exec(id);
+export function schedule_background_exec(): void {
+    ++pump_count;
+    globalThis.setTimeout(pump_message, 0);
+}
+
+export function mono_set_timeout(timeout: number, id: number): void {
+    function mono_wasm_set_timeout__() {
+        cwraps.mono_set_timeout_exec(id);
+    }
+    globalThis.setTimeout(mono_wasm_set_timeout__, timeout);
 }
 
 export function prevent_timer_throttling(): void {
-    if (isChromium) {
+    if (!isChromium) {
```

Comment:
😶

Reply:
Yep, just 7.0 problem introduced during the typescript refactoring.